### PR TITLE
Bg bindings result

### PR DESCRIPTION
**Pull request by sfc-gh-jslocum**
Added FDBResult to C bindings, and used it for readBlobGranules.

# Code-Reviewer Section

The general guidelines can be found [here](https://github.com/apple/foundationdb/wiki/FoundationDB-Commit-Process).

Please check each of the following things and check *all* boxes before accepting a PR.

- [ ] The PR has a description, explaining both the problem and the solution.
- [ ] The description mentions which forms of testing were done and the testing seems reasonable.
- [ ] Every function/class/actor that was touched is reasonably well documented.

## For Release-Branches

If this PR is made against a release-branch, please also check the following:

- [ ] This change/bugfix is a cherry-pick from the next younger branch (younger `release-branch` or `master` if this is the youngest branch)
- [ ] There is a good reason why this PR needs to go into a release branch and this reason is documented (either in the description above or in a linked GitHub issue)

### CODE REVIEW

File: fdbclient/ClientKnobs.cpp [hunk 1]
```diff
@@ -259,7 +259,8 @@ void ClientKnobs::initialize(Randomize randomize) {
 	init( MVC_CLIENTLIB_CHUNKS_PER_TRANSACTION,      32 );
 
 	// blob granules
-	init( ENABLE_BLOB_GRANULES,                   false );
+	// CHANGE BACK
```
Comment (sfc-gh-abeamon) [inline fdbclient/ClientKnobs.cpp]: Does this need to be undone before merging?

File: flow/ThreadHelper.actor.h [hunk 2]
```diff
@@ -727,5 +727,76 @@ class ThreadSafeAsyncVar : NonCopyable, public ThreadSafeReferenceCounted<Thread
 	ThreadSpinLock lock;
 };
 
+// Like a future (very similar to ThreadFuture) but only for computations that already completed. Reuses the SAV's
+// implementation for memory management  error handling though. Essentially a future that's returned from a synchronous
```
Comment (sfc-gh-abeamon) [inline flow/ThreadHelper.actor.h]:
```suggestion
// implementation for memory management error handling though. Essentially a future that's returned from a synchronous
```

File: flow/ThreadHelper.actor.h [hunk 3]
```diff
@@ -727,5 +727,76 @@ class ThreadSafeAsyncVar : NonCopyable, public ThreadSafeReferenceCounted<Thread
...
+	bool isValid() const { return sav != 0; }
+	bool isError() { return sav->isError(); }
+	Error& getError() {
+		if (!isError())
+			throw future_not_error();
+
+		return sav->error;
+	}
+
+	void cancel() { extractPtr()->cancel(); }
```
Comment (sfc-gh-abeamon) [inline flow/ThreadHelper.actor.h]: We may not need cancel here, I think it's essentially a no-op in this case.

File: flow/ThreadHelper.actor.h [hunk 4]
```diff
@@ -727,5 +727,76 @@ class ThreadSafeAsyncVar : NonCopyable, public ThreadSafeReferenceCounted<Thread
...
+
+	ThreadResult() : sav(0) {}
+	explicit ThreadResult(ThreadSingleAssignmentVar<T>* sav) : sav(sav) {
```
Comment (sfc-gh-abeamon) [inline flow/ThreadHelper.actor.h:749]: If we are going to allow construction via a `sav`, can we assert that the `sav` is in a ready state?

File: fdbclient/ClientKnobs.cpp (hunk 1 above)
Comment (sfc-gh-jslocum) [reply fdbclient/ClientKnobs.cpp]: Whoops. I unset this after but forgot to include that file

File: flow/ThreadHelper.actor.h (hunk 4 above)
Comment (sfc-gh-jslocum) [reply flow/ThreadHelper.actor.h:749]: Good idea

**Review by sfc-gh-abeamon — APPROVED** [review]

**Pull request merged**